lib/workflow/task-graph.js

```diff
@@ -0,0 +1,493 @@
+// Copyright 2015, EMC, Inc.
+'use strict';
+
+module.exports = taskGraphFactory;
+taskGraphFactory.$provide = 'TaskGraph.TaskGraph';
+taskGraphFactory.$inject = [
+    'Task.taskLibrary',
+    'TaskGraph.Store',
+    'Constants',
+    'Assert',
+    'uuid',
+    'Promise',
+    '_'
+];
+
+function taskGraphFactory(
+    taskLibrary,
+    store,
+    Constants,
+    assert,
+    uuid,
+    Promise,
+    _
+) {
+    function TaskGraph(definition, context, domain) {
```

Comment:
would you JSDoc the constructor details here? Would make this a lot more understandable down the road - what's in definition, context, and domain...